thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -0,0 +1,433 @@
+
+      const output = new Set(selectedUrns);
+      [...invisibleUrns].forEach(urn => output.delete(urn));
+      [...contextUrns].filter(urn => entities[urn] && entities[urn].type == 'metric').forEach(urn => output.add(urn));
+
+      return output;
+    }
+  ),
+
+  eventTableEntities: Ember.computed(
+    'entities',
+    'filteredUrns',
+    function () {
+      console.log('eventTableEntities()');
+      const { entities, filteredUrns } = this.getProperties('entities', 'filteredUrns');
+      return filterObject(entities, (e) => filteredUrns.has(e.urn));
+    }
+  ),
+
+  eventTableColumns: EVENT_TABLE_COLUMNS,
+
+  eventFilterEntities: Ember.computed(
+    'entities',
+    function () {
+      console.log('eventFilterEntities()');
+      const { entities } = this.getProperties('entities');
+      return filterObject(entities, (e) => e.type == 'event');
+    }
+  ),
+
+  tooltipEntities: Ember.computed(
+    'entities',
+    'invisibleUrns',
+    'hoverUrns',
+    function () {
+      const { entities, invisibleUrns, hoverUrns } = this.getProperties('entities', 'invisibleUrns', 'hoverUrns');
+      const visibleUrns = [...hoverUrns].filter(urn => !invisibleUrns.has(urn));
+      return filterObject(entities, (e) => visibleUrns.has(e.urn));
+    }
+  ),
+
+  isLoadingEntities: Ember.computed(
+    'entities',
+    '_pendingEntitiesRequests',
+    function () {
+      this.get('entities'); // force entities computation first
+      return this.get('_pendingEntitiesRequests').size > 0;
+    }
+  ),
+
+  isLoadingTimeseries: Ember.computed(
+    'timeseries',
+    '_pendingTimeseriesRequests',
+    function () {
+      this.get('timeseries'); // force timeseries computation first
+      return this.get('_pendingTimeseriesRequests').size > 0;
+    }
+  ),
+
+  //
+  // Entities loading
+  //
+
+  _entitiesLoader: Ember.computed(
+    'contextUrns',
+    'anomalyRegion',
+    'baselineRegion',
+    'analysisRegion',
+    function () {
+      console.log('_entitiesLoader()');
+      const { cache } = this.getProperties('_entitiesCache');
+      this._startRequestEntities();
+      return cache;
+    }
+  ),
+
+  _startRequestEntities() {
+    console.log('_startRequestEntities()');
+    const { contextUrns, anomalyRange, baselineRange, analysisRange, _pendingEntitiesRequests: pending } =
+      this.getProperties('contextUrns', 'anomalyRange', 'baselineRange', 'analysisRange', '_pendingEntitiesRequests');
+
+    const frameworks = ['relatedEvents', 'relatedDimensions', 'relatedMetrics'];
+
+    frameworks.forEach(framework => {
+      pending.add(framework);
+      const url = this._makeFrameworkUrl(framework, anomalyRange, baselineRange, analysisRange, contextUrns);
+      fetch(url)
+        .then(res => res.json())
+        .then(json => this._completeRequestEntities(this, json, framework));
+    });
+
+    this.setProperties({ _pendingEntitiesRequests: pending });
```

Comment:
You can use `this.set` instead here since you're only setting 1 property.

Reply:
uniformity